### PR TITLE
Zhirezhixin master

### PR DESCRIPTION
**Pull request by yifeili98**

### CODE REVIEW

**Comment by kis87988** [comment]
Please remember to rebase, the Merge branch commit should not be include.

**Comment by kis87988** [comment]
I purpose we should use Semantic Commit Messages.
Refer to https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716
and 
https://github.com/joelparkerhenderson/git_commit_message#begin-with-a-short-summary-line

File: ReadCourseData.py [hunk 1]
```diff
@@ -5,26 +5,58 @@ class Course:
 	def __init__(self, crn):
 		self.crn = crn
 
-	def __init__(self, crn, course_num, section_num,\
-		campus, num_credit, course_title, days, time, cap, act, wl_cap, wl_act, \
-		instructor_name, course_duration, location, attribute = '', lab_time = []):
+	def __init__(self, UID, crn, course_num, section_num,\
```
Comment (kis87988) [inline ReadCourseData.py]: Try to use `__init__(**kargs,*args)` style

File: ReadCourseData.py [hunk 2]
```diff
@@ -5,26 +5,58 @@ class Course:
...
+		self.department = list()
+		self.relatedCourse = dict()
+
+class Department:
+	def __init__(self, deptName):
+		self.deptName = deptName
+		self.courses = list()
+
+class Lab:
+	def __init__(self, UID, days, startTime, endTime, \
```
Comment (kis87988) [inline ReadCourseData.py]: same as above

File: ReadCourseData.py [hunk 3]
```diff
@@ -5,26 +5,58 @@ class Course:
...
+		self.startTime = startTime
+		self.endTime = endTime
+		self.startDate = startDate
+		self.endDate = endDate
+		self.instructor = instructor
+		self.location = location
+
+
 
 def main():
```
Comment (kis87988) [inline ReadCourseData.py:60]: we should remove path out here. This should not be absolutely path

File: ReadCourseData.py [hunk 4]
```diff
@@ -37,18 +69,24 @@ def main():
 			course_raw_data = json.load(f)
 
 	clist = []
```
Comment (kis87988) [inline ReadCourseData.py]: meaningful variable

File: ReadCourseData.py [hunk 5]
```diff
@@ -37,18 +69,24 @@ def main():
...
+	dlist = []
```
Comment (kis87988) [inline ReadCourseData.py]: meaningful variable

File: ReadCourseData.py [hunk 6]
```diff
@@ -37,18 +69,24 @@ def main():
...
 	for department in course_raw_data["2018 Winter De Anza"]["CourseData"]:
 		dep_course_num = 0
+		temp_dept = Department(department)
 		for c in course_raw_data["2018 Winter De Anza"]["CourseData"][department]:
 			dep_course_num += 1
-			temp_course = Course(c["CRN"], c["Crse"], c["Sec"], c["Cmp"], c["Cred"],
-							 c["Title"], c["Days"], c["Time"], c["Cap"], c["Act"], 
-							 c["WL Cap"], c["WL Act"], c["Instructor"], c["Date"],
-							 c["Location"], c["Attribute"], c["Lab Time"])
+			temp_course = Course(c["CRN"], c["CRN"], c["Crse"], c["Sec"], c["Cmp"], c["Cred"],
```
Comment (kis87988) [inline ReadCourseData.py]: try just pass c into the class. kargs should help

File: ReadCourseData.py [hunk 7]
```diff
@@ -37,18 +69,24 @@ def main():
...
+							 c["Title"], c["Days"], c["Time"][:8], c["Time"][9:], c["Cap"], 
+							 c["WL Cap"], c["Instructor"], c["Date"][:5], c["Date"][6:],
+							 c["Location"], c["Attribute"])
 			clist.append(temp_course)
+			interesting_name = temp_dept.deptName + " " + temp_course.course_num + " " + temp_course.course_title
```
Comment (kis87988) [inline ReadCourseData.py]: what is `interesting_name` mean?
try to use `"{} {} {}".format(var1,var2,var3)` style

**Comment by yifeili98** [comment]
咋rebase啊。。搜了下没看懂。。

**Review by kis87988 — CHANGES_REQUESTED** [review]
Please follow http://google.github.io/styleguide/pyguide.html python style guide.
A string should be either a single quote or double quote but not both. 

Another reference https://www.python.org/dev/peps/pep-0008/

File: ReadCourseData.py (hunk 1 above)
Comment (kis87988) [reply ReadCourseData.py]: since the parameters are too many. we should not list it all out.

**Comment by kis87988** [comment]
.env file should not be push to repo.
please reset and force push again. 
.evn should not in any of commit

**Pull request closed without merging**